### PR TITLE
Monitor in Chainstate

### PR DESCRIPTION
**Pull request by mrz1836**

### CODE REVIEW

**Comment by mrz1836** [comment]
@icellan is this ready to merge?

**Comment by icellan** [comment]
@mrz1836 Yes, I beleive so.

File: config/config.go [hunk 1]
```diff
@@ -95,6 +97,21 @@ type (
 		ServerPath     string `json:"server_path" mapstructure:"server_path"`         // server path i.e. "/graphql"
 	}
 
+	// MonitorOptions is the configuration for blockchain monitoring
+	MonitorOptions struct {
+		Debug                       bool    `json:"debug" mapstructure:"debug"`                                                 // true/false
+		Enabled                     bool    `json:"enabled" mapstructure:"enabled"`                                             // true/false
+		CentrifugeServer            string  `json:"centrifuge_server" mapstructure:"centrifuge_server"`                         // The server url address
```
Comment (mrz1836) [inline config/config.go:104]: Should have URL in the name to make it more clear

File: config/config.go [hunk 2]
```diff
@@ -95,6 +97,21 @@ type (
...
+		Token                       string  `json:"token" mapstructure:"token"`                                                 // Token to connect to the server with
```
Comment (mrz1836) [inline config/config.go:105]: Auth token?

File: config/envs/development.json [hunk 3]
```diff
@@ -31,6 +31,14 @@
     "transactions": false,
     "uri": "mongodb://localhost:27017/xapi"
   },
+  "monitor": {
```
Comment (mrz1836) [inline config/envs/development.json:34]: Is this missing from staging, prod and test json?

**Review by mrz1836 — COMMENTED** [review]
See my comments

**Comment by icellan** [comment]
@mrz1836 Wait, what?? I'm still working on your comments :-S

**Pull request merged**